mapie/utils.py

```diff
@@ -208,6 +209,37 @@ def check_cv(
         )
 
 
+def check_no_agg_cv(
+    cv: Union[int, str, BaseCrossValidator, BaseShuffleSplit],
+    no_agg_cv_array: list,
+) -> bool:
+    """
+    Check if cross-validator is ``"prefit"``, ``"split"`` or any split
+    equivalent `BaseCrossValidator` or `BaseShuffleSplit`.
+
+    Parameters
+    ----------
+    cv: Union[int, str, BaseCrossValidator, BaseShuffleSplit]
+        Cross-validator to check.
+
+    no_agg_cv_array: list
+        List of all non-aggregated cv methods.
+
+    Returns
+    -------
+    bool
+        True if `cv` is a split equivalent / non-aggregated cv method.
+    """
+    if isinstance(cv, str):
+        return cv in no_agg_cv_array
+    elif isinstance(cv, int):
+        return cv == 1
+    try:
+        return cv.get_n_splits() == 1
+    except Exception:
+        return False
```

Comment:
We should avoid using try/except conditions
```suggestion
    if hasattr(cv, "get_n_splits"):
        return cv.get_n_splits()
    else:
        return False    
```

Reply:
We should keep `return cv.get_n_splits() == 1` in this case.

Reply:
Indeed, I forgot it

Reply:
The other problem is that `get_n_splits` sometimes needs additional parameters. We have to throw an exception in all cases.

> E           TypeError: LeaveOneOut.get_n_splits() missing 1 required positional argument: 'X'

Reply:
@vincentblot28, I think I should keep the first proposal

Reply:
If we know the different possible cases, we must check them independantly, otherwise we do not clearly control what happens in the try/except condition (there are no try/except in MAPIE and we should keep it this way I think)

Reply:
We don't know all the possibilities, because users can create their own personalised CVs. Also, this would involve hard coding, which is harder to maintain. I suggest using try/except for good reason.

Reply:
In all cases, we must be shure of the number of split, I don't think it's a good behaviour to consider that the number of split is not equal to 1 as soon as `cv` has no attribut `get_n_split``

```suggestion
    if isinstance(cv, LeaveOneOut):
        return False
    if hasattr(cv, "get_n_split"):
        return cv.get_n_split() == 1 
    else:
        raise Exception("Your `cv` argument is not an `int` nore a `str` it should
        have a `get_n_split` method")
```

Reply:
After discussion, we're going to opt to externalise the "use_split_method" value with X dependency.